python/ql/src/experimental/semmle/python/security/injection/RegexInjection.qll

```diff
@@ -0,0 +1,53 @@
+/**
+ * Provides a taint-tracking configuration for detecting regular expression injection
+ * vulnerabilities.
+ */
+
+import python
+import experimental.semmle.python.Concepts
+import semmle.python.dataflow.new.DataFlow
+import semmle.python.dataflow.new.TaintTracking
+import semmle.python.dataflow.new.RemoteFlowSources
+
+/**
+ * A class to find methods executing regular expressions.
+ *
+ * See `RegexExecution`
+ */
+class RegexInjectionSink extends DataFlow::Node {
+  string regexModule;
+  Attribute regexMethod;
+
+  RegexInjectionSink() {
+    exists(RegexExecution reExec |
```

Comment:
Not necessarily feedback but more of an observation: 

So CodeQL is smart enough to find both the `DirectRegex` and `CompiledRegex` classes and run against them in the query. Is that because those classes inherit from `RegexExecution::Range`?

Reply:
Let's take it step by step

`RegexExecution::Range` is an abstract class, with concrete subclasses `DirectRegex` and `CompiledRegex`. So if you do `select any(RegexExecution::Range rr)`, you will get results from both `DirectRegex` and `CompiledRegex` -- so far, that's just normal CodeQL semantics.

The thing about having _both_ `RegexExecution` and ``RegexExecution::Range` is to allow for both extensibility and refinement, which is described in a bit more detail here: https://github.com/github/codeql/blob/main/docs/ql-design-patterns.md#range-for-extensibility-and-refinement

In general, I think questions like these are better suited for the Security Lab slack, or on https://github.com/github/securitylab/discussions (since other people can see the answers, and we don't end up cluttering PRs with general CodeQL questions)

Reply:
Thanks for the explanation @RasmusWL! :D

Reply:
Thanks @RasmusSemmle for the great explanation. I missed seeing the document on usage of range. 

You're right. This should be in slack. My bad on that one. Will not happen again.